Add Nature Order entry to options menu

miriamlefae opened the pull request:
A new entry for nature order was added to the options menu to let players choose the order in which natures appear in the multi-select .

## Description
Three options are provided: Alpha, Stat and Freq. Alpha is the default and the original alphabetical list
Stat lists the nature grouped and ordered by the stats they increase/decrease in the order of Atk > Def > Spe > SpA > SpD.
Freq orders the natures to put more useful ones at the top. This is the exact same order as Radical Red with the extra neutral natures tacked on at the end.

## Images
<img width="240" height="160" alt="pokeemerald-11" src="https://github.com/user-attachments/assets/97086a96-24d6-428d-8c50-59858ee50c84" /> <img width="240" height="160" alt="pokeemerald-12" src="https://github.com/user-attachments/assets/5ea2c767-a289-4283-b02f-6bc580aecec8" />
<img width="240" height="160" alt="pokeemerald-13" src="https://github.com/user-attachments/assets/8d35c1c0-ed62-4dd4-8b68-fc0f374dc242" /> <img width="240" height="160" alt="pokeemerald-14" src="https://github.com/user-attachments/assets/4205eda3-fff0-4a2b-9272-9571af4b37a2" />

## **Discord contact info**
kildemal

miriamlefae committed:
feat: add freq option to nature order

This is the exact same as radical red with the the extra neutrals tacked on at the bottom

miriamlefae committed:
fix: change if to elif in nature changer script

iriv24 commented on include/config/debug.h:
```diff
@@ -2,13 +2,13 @@
 // Overworld Debug
-#define DEBUG_OVERWORLD_MENU            FALSE                // Enables an overworld debug menu to change flags, variables, giving pokemon and more, accessed by holding R and pressing START while in the overworld by default.
-#define DEBUG_OVERWORLD_HELD_KEYS       (A_BUTTON)          // The keys required to be held to open the debug menu.
+#define DEBUG_OVERWORLD_MENU            TRUE                // Enables an overworld debug menu to change flags, variables, giving pokemon and more, accessed by holding R and pressing START while in the overworld by default.
```
undo changes here and in `general.h`

miriamlefae replied:
I saw that lol. Changed

miriamlefae committed:
feat: add option for nature order

miriamlefae committed:
fix: change initialSelected to fix stats mode

Stats mode had the default point to Admant which was not the first option. This fixes that

kildemal commented:
made one minor change so the initial selected in stats mode is lonely
